### DISCUSSION TITLE
Dutch language choices - keywords

### DISCUSSION THREAD

**Discussion by obw-go** [GrandOrgue/grandorgue#1969] [General]
In the language files /po/xx.po and /help/xx.po (where xx stands for a specific language) are a lot of keywords.

What would be the `best principles` to translate these keywords and what do those principles mean for the `best practice` for the translation of the language files?

In this discussion I like to deal with these questions. At first with some suggestions in a first setup, which will be expanded later at this place.

# Principles
1 Pure Dutch of mixed Dutch?
Try to use Dutch words and avoid loanwords from other languages?
Example: AI translated the keyword `loop` by` lus`. But when I look in a much used program like Audacity, in the Dutch version the word `loop` is used, so I change the translation back to the English loanword. Best choice?

2 Synonyms: yes or no?
Example: AI translates `manual` alternately with `manuaal`, `klavier` and `toetsenbord`. Should be made a strict choice for one of these translations or would alternate be allowed or even better?

3 When using AI translations, always systematic replacements have to be done. 
What are relevant rules for how to do these replacements?
Example 1 Think about the choice for replacing whole words only or words that can be parts of combined words: short words have a bigger chance to be also parts of other words, which would be changed as well when you don't choose for the replacement of "whole words". I first followed the AI translation of `loop `in Dutch bij `lus`. Later I went back to `loop`. If I would not have used "whole words" for the "replace all" command, also in words like "inclusief" and "plus" the "lus" part would have been replaced by "loop". Although "incloopief" and "ploop" have a nice ring to them, these neologisms would surprise the Dutch readers, even if they would have a hunch what went wrong...
Example 2 AI translates `organ` alternately with `orgel` and `orgaan`. In the context of GO the word `orgaan `is obviously wrong and by using this translation as well, the present AI translation-process proves to use a fairly limited amount of 'contextual awareness'. To be more precise: AI is not "aware" at all - it's just a statistical procedure that (based on training with examples) limits itself to text-chunks which are are kind-of-translated based on probability-relations between the chunks. Although the output of the process works most of the time out surprisingly well, even with a pretty technical text like the GO help file, every now and then (and then also repeatedly) the statistical procedures take the wrong turn...
For replacement of the wrong words is relevant: a. plural first (because in this case plurals are different: `orgels - organen`, b. singular second.

4 ..

# Practice

While I was working on the Dutch translation of the help file, I changed my mind about the translation of specific keywords, so I still have tuning work to do for both translation files. I suppose this will be the experience of all GO translators.

Here I like to make an alphabetic list of the keywords that could be translated differently, so where choices have to be made (or left open for alternative translations). 
Feedback welcome! Idealiter more GO-users giving feedback definitely would improve the translation choices.

`drawstop` registerknop trekregister
On most historic organs you have stops that are pulled and pushed. On electric organs you have drawstop buttons that are pushed twice for on/off. The translation `registerknop `is neutral to pull or push.
At first I used the word trekregister as translation of drawstop in the help file, but now I tend to reject this translation, so I plan to replace trekregister by registerknop.

`manual` manuaal klavier toetsenbord
Only use one or alternate?

`piston` piston registratieknop
Not to be confused with registerknop - `registratieknop `controls the setting of multiple stops (registers)
Only use one or alternate?

`rank` pijpenreeks pijpenrij register
Only use one or alternate?

`stop ` register registerknop
Only use one or alternate?

In the help file often the next combinations are used: "drawstop or button", "drawstop or pushbutton", "drawstop/button".
Should these words be seperated in the translation?
Or can they be translated with "(register-)knop"? 
I tend to prefer this translation.

**Comment by willeke1234** [comment]
Is this discussion about the Dutch translation or all translations?

I'm sorry but I don't understand principle 3.

**Reply by obw-go** [reply]
@willeke1234 
In this discussion it's about the Dutch translation.
As first step for the translation of the help file /help/po.nl I used an AI route as described in https://github.com/GrandOrgue/grandorgue/discussions/1950#discussion-6956089

The result of this AI route was a docx file with the translation of the strings in Dutch.
While working on the translation I encounter repeating translation errors in this docx file and instead of changing the same translation-errors over and again I use "replace all" for specific strings in that docx file. Which saves me time later when I copy/paste strings lower in the docx file from the docx file to the nl.po file in [poedit](https://poedit.net/) for final editting.

**Comment by willeke1234** [comment]
My opinion:

- Don't trust AI, it doesn't always handle synonyms and homonyms correctly. Always check and correct.
- The translation should use the vocabulary of the user of the software, in this case organists and sample set creators. Do they say klavier, manuaal or toetsenbord? For me, a toetsenbord is a keyboard. If you are not familiar with the jargon, you can try finding translations on websites such as www.pcorgan.com and Wikipedia.
- The English strings and text aren't perfect, this makes it more difficult to translate.
- General tips for translating:
	- be consistent, don't alternate synonyms, Imperative/infinitive, translated or not
    - take the context into account
	- check spelling
	- check spaces
	- check capitalization
	- check the .po file in a text editor, preferably a source-code editor or choose a monospaced font
	- test

Note: I'm a Mac user. I'm used to different standards.

**Reply by obw-go** [reply]
@willeke1234 

Good hints and tips!

Regarding you comment:
> * The English strings and text aren't perfect

Would it be an idea to start a discussion with the title "Improvement of the English GO strings and texts"?
Give it a flying start with the examples you found and your suggestions for improvement!

**Reply by willeke1234** [reply]
The whole GUI could be improved, but that's not a priority on Linux.
